### ISSUE TITLE
Filling holes with function arguments leads to Error : obligation cannot be used as a hint

### ISSUE THREAD

**Issue by kyoDralliam** [mattam82/Coq-Equations#193]
Taking a type (or a prop) as argument and an inhabitant `H` of this type and leaving a hole to be filled by `H` (either by automation, explicit obligation or proof mode with `Equation?`) leads to an error preventing the function to be defined. 
```Coq
From Equations Require Import Equations.
Fail Equations h_type (P : Type) (H : P) : P := h_type P H => let H1 := H in _.

(* h_type_obligations has type-checked, generating 1 obligation *)
(* Solving obligations automatically... *)
(* h_type_obligations_obligation_1 is defined *)
(* No more obligations remaining *)
(* h_type_obligations is defined *)
(* h_type is defined *)
(* h_type is defined *)
(* The command has indeed failed with message: *)
(* h_type_obligations_obligation_1 cannot be used as a hint. *)

Fail Equations h_prop (P : Prop) (H : P) : P := h_prop P H => _.
(* h_prop_obligations has type-checked, generating 1 obligation *)
(* Solving obligations automatically... *)
(* h_prop_obligations_obligation_1 is defined *)
(* No more obligations remaining *)
(* h_prop_obligations is defined *)
(* h_prop is defined *)
(* h_prop is defined *)
(* The command has indeed failed with message: *)
(* h_prop_obligations_obligation_1 cannot be used as a hint. *)
```
Using Coq 8.8.2 and Equations 1.2beta1

**Comment by mattam82** [comment]
Fixed by commit d5e08fc067c587f6c51e4f7ca49a4b7e7d70cd66 in master, I turned it into a warning instead.

**Comment by mattam82** [comment]
Probably not all user obligations should end up in the `f_wf_obligations` database though.
